backend/src/test/java/com/google/backend/servlets/AlertVisualizationServletTest.java

```diff
@@ -0,0 +1,108 @@
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito.*;
+import org.mockito.MockitoAnnotations;
+
+/** Contain tests for methods in {@link AlertVisualizationServlet} class. */
+@RunWith(JUnit4.class)
+public class AlertVisualizationServletTest {
+  @Mock HttpServletRequest request;
+  @Mock HttpServletResponse response;
+
+  @Rule // JUnit 4 uses Rules for testing specific messages
+  public ExpectedException thrown = ExpectedException.none();
+
+  private static final String RESPONSE_CONTENT_TYPE = "application/json;";
+  private static final String ID_PARAM = "id";
+  private static final Long FAKE_ID = 1L;
+  private static final String ENTITY_NOT_FOUND_ERROR = 
+    "com.google.appengine.api.datastore.EntityNotFoundException: "
+    + "No entity was found matching the key: alert(1)";
+
+  private static final AlertVisualizationServlet alertVisualizationServlet = new AlertVisualizationServlet();
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(
+    new LocalDatastoreServiceTestConfig());
+  private StringWriter stringWriter = new StringWriter();
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    when(response.getWriter()).thenReturn(new PrintWriter(stringWriter));
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doGet_ReturnsRequestedAlertEntity() throws IOException, ServletException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Alert newAlert = Alert.createAlertWithoutId(Timestamp.getDummyTimestamp(0), 
+        Arrays.asList(Anomaly.getDummyAnomaly()), Alert.StatusType.UNRESOLVED);
+    datastore.put(newAlert.toEntity());
+
+    Query query = new Query(Alert.ALERT_ENTITY_KIND);
+    Alert expectedAlert = Alert.createAlertFromEntity(datastore.prepare(query).asSingleEntity());
+    when(request.getParameter(ID_PARAM)).thenReturn(Long.toString(expectedAlert.getAlertId()));
+
+    
+    alertVisualizationServlet.doGet(request, response);
+
+    verify(response).setContentType(RESPONSE_CONTENT_TYPE);
+    JsonElement expected = JsonParser.parseString(
+      new Gson().toJson(expectedAlert));
+    JsonElement result = JsonParser.parseString(stringWriter.getBuffer().toString().trim());
+    assertEquals(expected, result);
+  }
+
+  @Test
+  public void doGet_AlertIdNotFound_ThrowsException() throws IOException, ServletException {
+    when(request.getParameter(ID_PARAM)).thenReturn(Long.toString(FAKE_ID));
+
+    thrown.expect(ServletException.class);
+    thrown.expectMessage(ENTITY_NOT_FOUND_ERROR);
```

Comment:
nit: it's ok to not test the alert message, since it might change from time to time.

Reply:
tested cause instead - thanks Paula for helping figure this out!